File: api/ZME.API.Shared/Models/Event.cs
using Microsoft.EntityFrameworkCore;

namespace ZME.API.Shared.Models;

[Index(nameof(Title))]
public class Event
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public required string Host { get; set; }
    public string? BannerUrl { get; set; }
    public DateTimeOffset Start { get; set; }
    public DateTimeOffset End { get; set; }
    public bool IsOpen { get; set; }
    public DateTimeOffset Created { get; } = DateTimeOffset.UtcNow;
    public DateTimeOffset Updated { get; set; } = DateTimeOffset.UtcNow;
}

File: ui/src/models/Event.ts

export type Event = {
    id: number;
    title: string;
    description: string;
    host: string;
    bannerUrl?: string;
    start: Date;
    end: Date;
    isOpen: boolean;
    created: Date;
    updated: Date;
};

export type EventPosition = {
    id: number;
    eventId: number;
    name: string;
    minRating: number;
    available: boolean;
}

export type EventRegistration = {
    id: number;
    userId?: number;
    eventId: number;
    eventPositionId: number;
    status: number
    start: Date;
    end: Date;
    created: Date;
    updated: Date;
}